hollow/src/main/java/com/netflix/hollow/api/consumer/metrics/AbstractRefreshMetricsListener.java

```diff
@@ -74,6 +78,23 @@ public final void blobLoaded(HollowConsumer.Blob transition) {
         updatePlanDetails.numSuccessfulTransitions ++;
     }
 
+    /**
+     * Metrics reporting implementation is provided by the extending subclass. If exceptions are not gracefully handled
+     * in the extending subclass then an exception there can fail the consumer refresh, even though metrics reporting
+     * might not be mission critical. This method protects against that scenario by catching all exceptions, logging
+     * that there was an exception, and continuing with the consumer refresh.
+     * @param refreshMetrics Consumer refresh metrics being reported
+     */
+
+    private final void noFailRefreshEndMetricsReporting(ConsumerRefreshMetrics refreshMetrics) {
+        try {
+            refreshEndMetricsReporting(refreshMetrics);
+        } catch (Exception e) {
+            // Metric reporting is not considered critical to consumer refresh. Log exceptions and continue.
+            log.log(Level.WARNING, "Encountered an exception in reporting consumer refresh metrics, ignoring exception and continuing with consumer refresh", e);
```

Comment:
should probably log at `ERROR` level. Even though we recover, it may represent a genuine problem.

Reply:
👍